### PR TITLE
Testing Your Ruby Code - Enumerable

### PR DESCRIPTION
**Pull request by Osong-Michael**
In this project I wrote code (Rspec), to test the behaviour of my enumerable methods and make sure they perform as intended.

### CODE REVIEW

File: spec/enum_spec.rb [hunk 1]
```diff
@@ -0,0 +1,156 @@
+require '../enum.rb'
+
```
Comment (majovanilla) [inline spec/enum_spec.rb]: Please use `require_relative` instead of `require`. It is a ruby convention.
Require should be used for external files, like gems, while require_relative should be used for referring to files within your directory.

File: spec/enum_spec.rb [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+RSpec.describe Enumerable do
+  let(:array1) { [0, 1, 2, 3, 4, 5, 6, 7, 8, 9] }
+  let(:array2) { %w[hello hey hi holla yea good] }
+  let(:array3) { [1, 2, 3, 4] }
+  let(:array4) { [2, 4, 6, 8] }
+  let(:array5) { [2, 4, 6, 8, 'jersey', %w[why me cr7 oa7]] }
+  let(:array6) { [4, 2, 3, 4, 4, 3, 5, 4, 4] }
+  let(:new_array) { [] }
```
Comment (majovanilla) [inline spec/enum_spec.rb:10]: You should include ranges as part of your testing variables.
Some methods behave differently when working with them and you need to make sure they are working! :smiley:

File: spec/enum_spec.rb [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+
+      expect(new_array).to eql(['1 : 2', '3 : 4'])
+    end
+  end
+
+  describe '#my_select' do
+    it 'Returns specific data from the array based on the argument passed' do
+      array2.my_select { |ele| new_array << ele if ele.include?('h') }
+      expect(new_array).to eq(%w[hello hey hi holla])
+    end
```
Comment (majovanilla) [inline spec/enum_spec.rb:57]: Please check the negative cases. When the array doesn't include the element you are looking for.

**Review by majovanilla — CHANGES_REQUESTED** [review]
Hey team!

Great test suite!! You covered almost everything!!

I have just minor requirements before approving your project:

- Check the commented code below and make the necessary changes.

- Try to include more negative test cases. I gave you an example in the comments. But the idea is this: look for edge cases, not the perfect test case. What happens if you give the wrong number of arguments? This will avoid unexpected behaviors in your methods and also in the future applications you build.

## Please, make all REQUIRED changes and submit your request again! 😀

If you have any questions, feel free to send me a message through Slack (@majo)
I would be happy to help! 👩‍💻 

Happy coding 🤓

--Maria Reyes

**Review by Torres-ssf — APPROVED** [review]
### Project Approved :muscle: 
Hi there :hand: 
Great job implementing the requested changes :clap: 
Both tests improvements were implemented (testing ranges and negative cases) :tada: 
**This PR is approved**
Good luck with the next project.

Happy coding :computer: 
**Sergio Torres**
